sdk/python/kfp/local/subprocess_task_handler_test.py

```diff
@@ -404,6 +404,26 @@ def my_comp(out_param: dsl.OutputPath(str),) -> int:
         self.assertEqual(task.outputs['out_param'], 'HelloHello')
         self.assertEqual(task.outputs['Output'], 1)
 
+    def test_outputpath_result_not_written(self):
+        local.init(runner=local.SubprocessRunner(use_venv=True))
+
+        @dsl.component
+        def my_comp(out_param: dsl.OutputPath(str)):
+            pass
+
+        task = my_comp()
+        self.assertEmpty(task.outputs)
+
+    def test_optional_param(self):
+        local.init(runner=local.SubprocessRunner(use_venv=True))
+
+        @dsl.component
+        def my_comp(string: Optional[str] = None):
+            assert string is None
```

Comment:
This component by itself is wrong--the input parameter cannot accept any other input values. 
I think a more proper way to test optional default value would be move assert outside component implementation:
```python
@dsl.component
def my_comp(string: Optional[str] = None) -> str:
    return 'default' if string is None else 'non-default'

task = my_comp()
self.assertEqual(task.outputs, 'default')
```

Reply:
Agreed. Fixed.